configure.ac

```diff
@@ -467,6 +467,25 @@ if test "$no_pie_option_available" = "yes"; then
 fi
 AC_SUBST([NO_PIE_LDFLAG])
 
+dnl ***************************************************************************
+dnl Set up LDFLAGS_FORCE_FLAT_NAMESPACE
+
+
+if test $ostype = "Darwin"; then
+  LDFLAGS_FORCE_FLAT_NAMESPACE="-Wl,-force_flat_namespace"
```

Comment:
I asked the same today. MacOS has a so called [two-level namespace linking](https://developer.apple.com/library/archive/documentation/Porting/Conceptual/PortingUnix/compiling/compiling.html#//apple_ref/doc/uid/TP40002850-BCIHJBBF) by default.
That prevents defining your own implementation for functions that are present in other libs. (like supplying your own `read` function).

That is a good feature by default, but for test it can prevent stubbing technics. The new test case uses it to fake time:
 https://github.com/balabit/syslog-ng/pull/2427/files#diff-e45df9d43f2cc8ab23500d13ac59e5e2R32

Reply:
As @Kokan explained, it allows a unit test to inject function implementations defined in other shared libraries. the timestamp parsing test uses it to fake the current time by providing a cached_g_current_time() function.

https://www.unix.com/man-page/osx/1/ld/

The default behavior of the linker on macOS differs from what Linux does, this switch restores that behavior.